config/resource.go

```diff
@@ -397,6 +405,13 @@ func (config *Service) Validate(path string) ([]string, error) {
 		golog.Global().Debugw("no name given, defaulting name to builtin")
 		config.Name = resource.DefaultServiceName
 	}
+	matched, err := regexp.MatchString(`^[a-zA-Z][-\w]*$`, config.Name)
```

Comment:
same here. can pull this out to a common function

Reply:
done